### PR TITLE
Add support for github actions

### PR DESCRIPTION
**Pull request by shubham149**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,3 +26,24 @@ Execute a Bitrise plugin:
...
+        who-to-greet: Mona the Octocat
+    env:
+        hello: world
+```
+
+```
+export PLUGIN_WITH="{ \"who-to-greet\": \"Mona the Octocat\" }"
+export hello=world
+
+plugin -type action -name actions/hello-world-javascript-action@v1.1 -repo https://github.com/actions/hello-world-javascript-action.git
```
Comment (bradrydzewski) [inline README.md]: is `-repo` required here, or can we just use `-name` for github?

File: pkg/encoder/encoder.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
+package encoder
```
Comment (bradrydzewski) [inline pkg/encoder/encoder.go]: can we move to `plugin/internal/` or will other programs need to import this package?

File: plugin/github/env.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+			return nil, errors.Wrap(err, "with attribute is not of map type with key & value as string")
+		}
+
+		return with, nil
+	}
+	return nil, nil
+}
+
+func GetEnv(envVars map[string]string) map[string]string {
+	dst := make(map[string]string)
```
Comment (bradrydzewski) [inline plugin/github/env.go:31]: can we make `GetWith` and `GetEnv` private?

File: plugin/github/execer.go [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+// Exec executes a github action.
+func (e *Execer) Exec(ctx context.Context) error {
+	envVars := environ.Map(e.Environ)
+	tmpDir, err := ioutil.TempDir("", "")
+	if err != nil {
+		return err
+	}
+	workflowFile := fmt.Sprintf("%s/workflow.yml", tmpDir)
+
+	if err := createWorkflowFile(workflowFile, e.Name, envVars); err != nil {
```
Comment (bradrydzewski) [inline plugin/github/execer.go]: use `filepath.Join` for windows compatibility?

File: plugin/github/execer.go [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+		"-W",
+		workflowFile,
+		"-P",
+		fmt.Sprintf("-self-hosted=-self-hosted"),
+		"-b",
+		"--detect-event",
+	}
+
+	if eventPayload, ok := envVars["PLUGIN_EVENT_PAYLOAD"]; ok {
+		eventPayloadFile := fmt.Sprintf("%s/event.yml", tmpDir)
```
Comment (bradrydzewski) [inline plugin/github/execer.go]: use `filepath.Join` for windows compatibility?

**Comment by bradrydzewski** [comment]
looks good. left some minor comments.

one question I have is if we need to support these environment variables:
https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables

for the bitrise plugin we map all the harness / drone environment variables to the bitrise types:
https://github.com/drone/plugin/blob/main/plugin/bitrise/environ.go#L40

**Comment by shubham149** [comment]
@bradrydzewski nektos/act sets the github environment variables. We are passing the webhook event payload json file which is parsed by the nektos to set the variables

**Pull request merged**